src/pages/stats/[login].tsx

```diff
@@ -22,6 +22,27 @@ export default function Stats() {
     repositories: PullRequestContributionsByRepository[];
     isLoading: boolean;
   } = useGitHubPullRequests(year, login as string);
+  const [searchQuery, setSearchQuery] = useState<string>("");
+
+  const filteredRepositories = useMemo(() => {
+    // putting the default value of the repositories
+    if (!searchQuery) {
+      return repositories;
+    }
+
+    // getting search query
+    const query = searchQuery.toLowerCase();
+    const filterRepos = repositories.filter((repoData) =>
+      repoData.repository.name.toLowerCase().includes(query)
+    );
+
+    // checking if there is no repositories
+    if (filterRepos.length === 0) {
+      return [];
+    }
+
+    return filterRepos;
```

Comment:
Do we need to manually check if there are no repos?

`return repositories.filter(...)` should be enough

Reply:
Yeah, there is no need for that conditional statement. Thanks for the review. Just updated the code